lib/Component.php

```diff
@@ -238,7 +238,7 @@ public function select($name)
                 return array_filter(
                     $result,
                     function ($child) use ($group) {
-                        return $child instanceof Property && strtoupper($child->group) === $group;
+                        return $child instanceof Property && strtoupper((string) $child->group) === $group;
```

Comment:
I would prefer an explicit null check instead of a cast

Reply:
it would also be usefull if we could come up with a more fine grained type then `array` for the `children` property

Reply:
I actually used the case because this is also done in other places in the existing code and IMO it's the best readable option when still needing to support PHP 5.5 as you intend to by the README. Anyway, I changed the code to match your preference.
I also adapted the array type, also I don't see how it helps here. What could help is declaring the `$group` property as nullable in the Property class, but phpstan does not seem to care, maybe because it is set to a very basic level.